sky/shell/platform/mojo/platform_view_mojo.cc

```diff
@@ -147,5 +156,23 @@ void PlatformViewMojo::OnEvent(mojo::EventPtr event,
   callback.Run();
 }
 
+void PlatformViewMojo::Create(
+    mojo::ApplicationConnection* connection,
+    mojo::InterfaceRequest<::keyboard::KeyboardService> request) {
+
+  mojo::ServiceProviderPtr sp;
+  connector_->ConnectToApplication("mojo:keyboard", mojo::GetProxy(&sp), nullptr);
+
+#ifdef OS_LINUX
+  ::keyboard::KeyboardServiceFactoryPtr factory;
+  mojo::NativeViewportEventDispatcherPtr dispatcher;
+  sp->ConnectToService(::keyboard::KeyboardServiceFactory::Name_, mojo::GetProxy(&factory).PassMessagePipe());
```

Comment:
There's a `mojo::ConnectToService` helper in `mojo/public/cpp/application/connect.h` that does most of this work for you:

```
mojo::ConnectToService(keyboard_service_provider.get(), &factory);
```